streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```
@@ -162,15 +163,39 @@ public static boolean isPermitted(
             }
             final R result = (R) iterator;
             return QueryResult.forResult(result);
-        } catch (final Throwable t) {
-            final String message = parseStoreException(t, store, query);
+        } catch (final Exception e) {
```

Changed from Throwable to Exception to avoid swallowing Errors